[bug] Build error on macOS ARM with libsamplerate-sys v0.1.12

**describe the bug**
I'm unable to build the screenpipe project from source on macOS (Apple Silicon) due to a build failure in the libsamplerate-sys v0.1.12 crate. The build fails with a CMake error about outdated compatibility settings:


```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

```

**system info**
- os: macos 15.3.1, M2
- screenpipe version: 0.2.74
- cmake version: 4.0.0

**additional context**

this is the error:
```
error: failed to run custom build command for `libsamplerate-sys v0.1.12`

Caused by:
  process didn't exit successfully: `/Users/abhishekpathak/Documents/codebase/rust/opensource/screenpipe/target/release/build/libsamplerate-sys-edbeaba59e0260a2/build-script-build` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_aarch64-apple-darwin = None
  CMAKE_TOOLCHAIN_FILE_aarch64_apple_darwin = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_aarch64-apple-darwin = None
  CMAKE_GENERATOR_aarch64_apple_darwin = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_aarch64-apple-darwin = None
  CMAKE_PREFIX_PATH_aarch64_apple_darwin = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_aarch64-apple-darwin = None
  CMAKE_aarch64_apple_darwin = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/Users/abhishekpathak/Documents/codebase/rust/opensource/screenpipe/target/release/build/libsamplerate-sys-eed87b2b4cc1a125/out/build" && CMAKE_PREFIX_PATH="" LC_ALL="C" "cmake" "/Users/abhishekpathak/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libsamplerate-sys-0.1.12/libsamplerate" "-DLIBSAMPLERATE_TESTS=OFF" "-DLIBSAMPLERATE_EXAMPLES=OFF" "-DLIBSAMPLERATE_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DCMAKE_INSTALL_PREFIX=/Users/abhishekpathak/Documents/codebase/rust/opensource/screenpipe/target/release/build/libsamplerate-sys-eed87b2b4cc1a125/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx -mmacosx-version-min=15.4" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx -mmacosx-version-min=15.4" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx -mmacosx-version-min=15.4" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=MinSizeRel"
  -- Configuring incomplete, errors occurred!

  --- stderr
  CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 3.5 has been removed from CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.

    Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.



  thread 'main' panicked at /Users/abhishekpathak/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cmake-0.1.54/src/lib.rs:1119:5:

  command did not execute successfully, got: exit status: 1

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```


**Things I've tried**
- Reinstalled cmake via Homebrew
- Set `CMAKE_PREFIX_PATH`
- Installed libsamplerate via brew
- Tried rebuilding with clean target

**What I need help with**
- Is this a known issue with libsamplerate-sys?
- Any workaround for Apple Silicon?
- Should we pin a specific cmake version?

--- a/screenpipe-audio/Cargo.toml
+++ b/screenpipe-audio/Cargo.toml
@@ -88,8 +88,8 @@ ort = { version = "=2.0.0-rc.6", features = [
   "cuda",
 ] }
 esaxx-rs = "0.1.10"
-samplerate = { version = "0.2.4" }
-libsamplerate-sys = "0.1.10"
+samplerate = { git = "https://github.com/divanshu-go/rust-samplerate.git", rev = "bf8415d5ac33940bb5a52bd23b24ca9a56f96792" }
+libsamplerate-sys = { git = "https://github.com/divanshu-go/libsamplerate-sys.git", rev = "6545fdd165673fce30d4ed5e82b845adc7b2faac" }
 
 [target.'cfg(target_os = "macos")'.dependencies]
 once_cell = "1.17.1"

